src/plots/cartesian/axis_defaults.js

```diff
@@ -155,10 +160,55 @@ function rangebreaksDefaults(itemIn, itemOut, containerOut) {
 
     if(enabled) {
         var bnds = coerce('bounds');
-
         if(bnds && bnds.length >= 2) {
-            if(bnds.length > 2) {
-                itemOut.bounds = itemOut.bounds.slice(0, 2);
+            var dfltPattern = '';
+            var i, q;
+            if(bnds.length === 2) {
+                for(i = 0; i < 2; i++) {
+                    q = indexOfDay(bnds[i]);
+                    if(q) {
+                        dfltPattern = DAY_OF_WEEK;
+                        break;
+                    }
+                }
+            }
+            var pattern = coerce('pattern', dfltPattern);
+            if(pattern === DAY_OF_WEEK) {
+                for(i = 0; i < 2; i++) {
+                    q = indexOfDay(bnds[i]);
+                    if(q) {
+                        // convert to integers i.e 'Sunday' --> 0
+                        itemOut.bounds[i] = bnds[i] = q - 1;
+                    }
+                }
+            }
+            if(pattern) {
+                // ensure types and ranges
+                for(i = 0; i < 2; i++) {
+                    q = bnds[i];
+                    switch(pattern) {
+                        case DAY_OF_WEEK :
+                            if(isNumeric(q)) q = Math.floor(q);
+
+                            q = Math.floor(+q);
```

Comment:
redundant - and dangerous - `q=''` would be turned into `0` for example. We should only cast to number when the value has already passed `isNumeric` (but might be a numeric string).

Also I don't think we should `floor`, we should just discard values that aren't integers (ie `enabled=false`). For one thing, that way if we ever add support for fractional days it won't break any existing behavior.

Can't this numeric validation be pushed into the `if(pattern === DAY_OF_WEEK)` loop above, and only executed if we didn't already convert from a day name to integer?

Reply:
Good call. Fixed in https://github.com/plotly/plotly.js/pull/4661/commits/9a0416aefb17fd0d5ac3bd26f91abb721b934a6d.